pages/food-groups/index.tsx

```diff
@@ -0,0 +1,280 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import { useRef } from 'react'
+import FoodGroups from 'components/FoodGroups'
+
+import Auth from '@components/Auth'
+import { Button, Modal } from '@components/Custom'
+import { ACHIEVEMENT, useFirebase } from '@components/FirebaseContext'
+import { DRAGGING_STATE } from '@components/FoodGroups/Draggable'
+import {
+  ORIGIN_VECTOR2,
+  Vector2
+} from '@components/FoodGroups/Draggable/boundingbox'
+import {
+  CHARACTER_POSITIONS,
+  CharacterSpawner,
+  generateCharacterSet
+} from '@components/FoodGroups/Draggable/characterspawner'
+import {
+  FoodGroupCharacterImage,
+  FoodGroupCharacterImageDynamic
+} from '@components/FoodGroups/Draggable/types'
+import { GROUPS } from '@components/FoodGroups/groups'
+import {
+  draggableZoneStyle,
+  startZoneStyle
+} from '@components/FoodGroups/styles'
+import { State_ } from '@components/FoodGroups/types'
+import {
+  getCharacterData,
+  getFormatData
+} from '@components/NotionAPI/getCharacterData'
+
+const N_DRAGGABLE = 5
+const newArray = (v: any) => Array(N_DRAGGABLE).fill(v)
+
+interface Props {
+  notion_character_data: FoodGroupCharacterImage[]
+}
+
+const FoodGroupsPage: React.FC<Props> = ({ notion_character_data }: Props) => {
+  const { achievements, updateAchievementsDocument, user } = useFirebase()
+  // SIGN IN FORM
+  const [openSignInForm, setOpenSignInForm] = useState(false)
+  const toggleOpenSignInForm = useCallback(() => {
+    setOpenSignInForm(prev => !prev)
+  }, [])
+
+  const [modalState, setModalState] = useState(false)
+  const [selectedDraggableType, setSelectedDraggableType] = useState(
+    GROUPS.NONE
+  )
+
+  // GAME STATE
+  const [hoverType, setHoverType] = useState(GROUPS.DEFAULT)
+  const [roundCounter, setRoundCounter] = useState(0)
+  const [correctDraggables, setCorrectDraggables] = useState(() =>
+    newArray(false)
+  )
+  const [wheelEnabled, setWheelEnabled] = useState(true)
+
+  // This keeps the number of wins while signed out so when the user logs in it gets added to their score
+  const [roundCounterSignedOut, setRoundCounterSignedOut] = useState(0)
+
+  // Draggable Logic
+  const [overridePosition, setOverridePosition] = useState(ORIGIN_VECTOR2)
+
+  const [currentCharSet, setCharSet] = useState<
+    FoodGroupCharacterImageDynamic[]
+  >(() => generateCharacterSet(notion_character_data))
+  const [nextCharSet, setNextCharSet] = useState<
+    FoodGroupCharacterImageDynamic[]
+  >(() => generateCharacterSet(notion_character_data))
+
+  const draggablePositions_1: State_<Vector2>[] = []
+  const draggablePositions_2: State_<Vector2>[] = []
+  for (let i = 0; i < 5; i++) {
+    draggablePositions_1[i] = useState<Vector2>(() => ORIGIN_VECTOR2)
+    draggablePositions_2[i] = useState<Vector2>(() => ORIGIN_VECTOR2)
+  }
```

Comment:
Consider refactoring this section as I think there is another way of doing this without `eslint` complaining.